### PR TITLE
Added checks for clang::Cursor::template_arg_value

### LINKED ISSUE #134: clang::Cursor::template_arg_value should behave defensively
We should check that the cursor's referent is a template specialization and that `i` is not out of bounds. If either of those things don't hold, we should return `None` and avoid the FFI call. There is no way to check the return value for validity if we mess this up: the behavior is undefined! Otherwise, we should make the FFI call and return `Some`.

This would be a nice removal of a foot gun.

http://clang.llvm.org/doxygen/group__CINDEX__TYPES.html#ga46e363545effaa0794a2ba4bcfae1fe3

> It is undefined to call this function on a CXCursor that does not represent a FunctionDecl or whose I'th template argument is not an integral value.

I can mentor whoever would like to work on this bug.

### PR DESCRIPTION
**Pull request by cynicaldevil**
Fixes #134.
r? @fitzgen

### CODE REVIEW

**Comment by emilio** [comment]
Is there any caller in this function? If not, let's rip it out.

File: src/clang.rs [hunk 1]
```diff
@@ -457,8 +457,26 @@ impl Cursor {
...
+                        true
+                    }
+                    else {
+                        false
+                    }
+                },
+            None => false
+        };
+        match within_bounds {
+            true => {
```
Comment (fitzgen) [inline src/clang.rs]: In general, using `if` is usually better than `match` for boolean values:

```rust
if within_bounds {
    ...
} else {
    ...
}
```

**Comment by cynicaldevil** [comment]
@fitzgen I've removed the method and updated the PR. Think I'll pick up another issue to work on as well, since this one turned out to be really trivial.

**Comment by cynicaldevil** [comment]
@Wafflespeanut I had forgotten to to run the build step after removing the method. All good now!

**Pull request merged**